examples/main/main.cpp

```diff
@@ -167,26 +167,35 @@ int main(int argc, char ** argv) {
     std::vector<llama_token> session_tokens;
 
     if (!path_session.empty()) {
-        fprintf(stderr, "%s: attempting to load saved session from '%s'\n", __func__, path_session.c_str());
-
-        // fopen to check for existing session
-        FILE * fp = std::fopen(path_session.c_str(), "rb");
-        if (fp != NULL) {
-            std::fclose(fp);
-
-            session_tokens.resize(params.n_ctx);
-            size_t n_token_count_out = 0;
-            if (!llama_load_session_file(ctx, path_session.c_str(), session_tokens.data(), session_tokens.capacity(), &n_token_count_out)) {
-                fprintf(stderr, "%s: error: failed to load session file '%s'\n", __func__, path_session.c_str());
-                return 1;
-            }
-            session_tokens.resize(n_token_count_out);
-            llama_set_rng_seed(ctx, params.seed);
+        auto load_session = [&]() {
+            fprintf(stderr, "load_session: attempting to load saved session from '%s'\n" , path_session.c_str());
+
+            // fopen to check for existing session
+            FILE * fp = std::fopen(path_session.c_str(), "rb");
+            if (fp != NULL) {
```

Comment:
You can simplify the control flow further if you invert the condition on this 'if' statement, and return so you don't need the 'else' clause:
```cpp
if (fp == NULL) {
    fprintf(stderr, "load_session: session file does not exist, will create\n");
    return;
}
```

Reply:
I reworked the control flow, thanks!